feat(cal): add hover for day

igorashs opened the pull request:
- added hovered state/style for calendar day
- fixed "other" days (offsets) - the 'selected' class shouldn't not be applied

demos:
- both directions
![Screenshot 2023-12-27 at 19 00 11](https://github.com/k5cjs/k5cjs/assets/39920127/ad2a1a09-56e9-4041-9248-b08d303f3360)
![Screenshot 2023-12-27 at 19 00 30](https://github.com/k5cjs/k5cjs/assets/39920127/eff451d2-57b8-408d-9d90-f9d755fbe9dd)
![Screenshot 2023-12-27 at 19 00 45](https://github.com/k5cjs/k5cjs/assets/39920127/03c85810-608a-485d-ad8f-f8cd076255ed)

- "offset" days can be clicked to select next/prev month days
![Screenshot 2023-12-27 at 19 00 58](https://github.com/k5cjs/k5cjs/assets/39920127/98383a4d-5557-4316-9b90-4ab95261037f)

kryshac commented on projects/cal/src/lib/components/kc-cal-day/kc-cal-day.component.ts:
```diff
@@ -97,13 +101,41 @@ export class KcCalDayComponent<T extends KcCalSelector = KcCalSelector> implemen
   }
 
+  @HostBinding('class.kc-cal-day--hovered')
+  get hovered() {
```
add return type

kryshac commented on projects/cal/src/lib/components/kc-cal-day/kc-cal-day.component.ts:
```diff
@@ -97,13 +101,41 @@ export class KcCalDayComponent<T extends KcCalSelector = KcCalSelector> implemen
+    if (!from && !to) return this._isHovered;
+
+    // when 'to' is not set, hover when is between from & hovered dates
+    if (!!from && !to)
+      return this._isDateBetween(this.day, from, hovered) || this._isDateBetween(this.day, hovered, from);
+
+    // hover when is between either from/to & hovered dates
+    return (
+      (!!from && this._isDateBetween(this.day, from, hovered)) || (!!to && this._isDateBetween(this.day, hovered, to))
+    );
```
add tests for all cases.

A case that I don't think will work:
The user first sets the to date and then selects a form that is greater than to

![Peek 2023-12-28 11-28](https://github.com/k5cjs/k5cjs/assets/19764133/3a340736-ac47-4f46-a46c-93808292b801)

kryshac commented on projects/cal/src/lib/components/kc-cal-day/kc-cal-day.component.scss:
```diff
@@ -26,7 +26,6 @@
 
   &--other {
     color: #94a3b8;
-    cursor: default;
```
why did you remove it?

kryshac replied:
if you want the user to be able to click on the days of the previous and the following month, then implement in another pull request because there are more cases that need to be implemented

igorashs replied:
to show the pointer cursor instead, you can click on "other" (offsets) days

kryshac replied:
continuation of the above message

1 When you click on a day that does not belong to the current month, then the month must also change
2 Hover will have to work on them too, just with a different shade
3 And also the selected range must be seen but also with a different shade

kryshac replied:
![Peek 2023-12-28 11-45](https://github.com/k5cjs/k5cjs/assets/19764133/7dfa142a-8f41-47b0-850a-f0f322095a16)

igorashs commented:
Update:
- added unit test coverage
- added distinct style for selected & hovered offset days
- added new use case for hovered binding "when 'from' is not set, hover when is between to & hovered dates" 

demos:
![Screenshot 2023-12-28 at 15 12 10](https://github.com/k5cjs/k5cjs/assets/39920127/b4f2d602-fdf5-4b42-a212-e04f6782e282)

![Screenshot 2023-12-28 at 15 12 30](https://github.com/k5cjs/k5cjs/assets/39920127/d0e59819-aa4d-4339-8b64-4b722da23dba)

![Screenshot 2023-12-28 at 15 22 05](https://github.com/k5cjs/k5cjs/assets/39920127/94cfb654-bc0a-4df9-83a2-8136b6d6f860)

kryshac commented:
![image](https://github.com/k5cjs/k5cjs/assets/19764133/2d32bcf2-0fb4-46d0-90a2-df9a3b4335a8)
the round edges must be fixed

igorashs commented:
update:
 - fixed round edges taking hovered in consideration
 
[screen-recorder-fri-dec-29-2023-11-38-12.webm](https://github.com/k5cjs/k5cjs/assets/39920127/503966ba-ac1c-45e2-a583-95bd84ae93f7)
